src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,426 @@
+                __METHOD__,
+                $this->options['gcProbablity']
+            ));
+        }
+
+        if ($this->initialTtl <= 0) {
+            throw new InvalidArgumentException(sprintf(
+                '%s() expects a strictly positive TTL. Got %d.',
+                __METHOD__,
+                $this->initialTtl
+            ));
+        }
+    }
+
+    /**
+     * Create a TTL index to automatically remove expired locks.
+     *
+     * If the gcProbablity option is set higher than 0.0 (defaults to 0.001);
+     * there is a chance this will be called on self::save().
+     *
+     * Otherwise; this should be called once manually during database setup.
+     *
+     * Alternatively the TTL index can be created manually on the database:
+     *
+     *  db.lock.ensureIndex(
+     *      { "expires_at": 1 },
+     *      { "expireAfterSeconds": 0 }
+     *  )
+     *
+     * Please note, expires_at is based on the application server. If the
+     * database time differs; a lock could be cleaned up before it has expired.
+     * To ensure locks don't expire prematurely; the lock TTL should be set
+     * with enough extra time to account for any clock drift between nodes.
+     *
+     * A TTL index MUST BE used on MongoDB 2.2+ to automatically clean up expired locks.
+     *
+     * @see http://docs.mongodb.org/manual/tutorial/expire-data/
+     *
+     * @throws UnsupportedException          if options are not supported by the selected server
+     * @throws MongoInvalidArgumentException for parameter/option parsing errors
+     * @throws DriverRuntimeException        for other driver errors (e.g. connection errors)
+     */
+    public function createTtlIndex(int $expireAfterSeconds = 0)
+    {
+        if ($this->getDatabaseVersion() < '2.2') {
+            throw new BadMethodCallException('Database version does not support TTL Indexes');
+        }
+
+        $keys = [
+            'expires_at' => 1,
+        ];
+
+        $options = [
+            'expireAfterSeconds' => $expireAfterSeconds,
+        ];
+
+        $this->getCollection()->createIndex($keys, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws LockExpiredException when save is called on an expired lock
+     */
+    public function save(Key $key)
+    {
+        $key->reduceLifetime($this->initialTtl);
+
+        try {
+            $this->upsert($key, $this->initialTtl);
+        } catch (WriteException $e) {
+            if ($this->isDuplicateKeyException($e)) {
+                throw new LockConflictedException('Lock was acquired by someone else', 0, $e);
+            }
+            throw new LockAcquiringException('Failed to acquire lock', 0, $e);
+        }
+
+        if ($key->isExpired()) {
+            throw new LockExpiredException(sprintf('Failed to store the "%s" lock.', $key));
+        }
+
+        if ($this->options['gcProbablity'] > 0.0
+            && (
```

Comment:
this CS is not following symfony CS

Reply:
I'm used to PSR-1 + PSR-2, i've updated to Symfony style